server.go

```diff
@@ -0,0 +1,55 @@
+package main
```

Comment:
I know it's a WIP, just taking notes: all docstrings for the package and functions are missing.